cmd/pipeline/deploy.go

```diff
@@ -234,8 +236,21 @@ func (pc *Command) waitUntilRunning(ctx context.Context, name string, action *ty
 	signal.Notify(stop, os.Interrupt)
 	exit := make(chan error, 1)
 
-	go func() {
+	var wg sync.WaitGroup
```

Comment:
If I'm reading this right, this leaks a go routine if an error is returned.

We have `exit := make(chan error, 1)` which has a buffer of 1. This means that it can hold 1 message without blocking. Before this change it was fine because we sent only one error from either `waitToBeDeployed` or `waitForResourcesToBeRunning` (lines 241 and 246).

With this change, we are sending two messages to the exit channel so the second one will block until we have a receiver which never happens.

Also, it looks like we are doing a `Wait()` right before selecting on `<-stop` so the CTRL+C will probably not work.

In general we should make better use of the context and use it as a cascading cancellation method. I don't think we should even call `signal.Notify(stop, os.Interrupt)` here and just use the context and cancel everything when done.

Ideally, we should only have a single top level `signal.Notify(stop, os.Interrupt)` at the root command and cascade a context cancellation. Not in scope for this PR though.

Created this gist as a reference implementation in case it helps and you wanna take a look: https://gist.github.com/maroshii/0a4e0f402d1d902e5491b3eaba108ff9. It has both a context cancellation method and the signal the way it is here.

It might require changes to the other functions to receive a context and have proper cancellation though